awx/main/models/events.py

```diff
@@ -520,13 +520,21 @@ def _update_host_summary_from_stats(self, hostnames):
                 (summary['host_id'], summary['id'])
                 for summary in JobHostSummary.objects.filter(job_id=job.id).values('id', 'host_id')
             )
+            updated_hosts = set()
             for h in all_hosts:
```

Comment:
okay, so `host_map` is _not_ down-selected because it's sent over the wire as an artifact from the inventory generation. Is that right? Then `all_hosts` follows from that, and the only time we get what the limited hosts were are from the stats event, which comes through in `host_mapping`.

`hostnames` comes from...

https://github.com/ansible/awx/blob/de5f996358563eb900645d3620ace82e77451ad7/awx/main/models/events.py#L476-L483

Note that, as I write that, I see clear as day that its criteria is different from L504 above

```python
                for stat in ('changed', 'dark', 'failures', 'ignored', 'ok', 'processed', 'rescued', 'skipped'):
```

which is not what you came here to fix... but...

Reply:
Oh you know what, that might actually be https://github.com/ansible/ansible-runner/issues/196